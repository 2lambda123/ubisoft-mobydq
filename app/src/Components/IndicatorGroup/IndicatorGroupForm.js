import React from 'react';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import TextInput from './../FormInput/TextInput';
import Button from '@material-ui/core/Button';
import SaveIcon from '@material-ui/icons/Save';

const IndicatorGroupFormFields = (props) => {
  const {
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting
  } = props;
  return <form onSubmit={handleSubmit} style={{ 'marginLeft': '60px' }}>
    <div>
      <TextInput
        id="name"
        label="Name"
        helperText=""
        placeholder=""
        touched={touched.name}
        error={touched.name && errors.name}
        value={values.name}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="createdDate"
        label="Created Date"
        helperText=""
        placeholder=""
        touched={touched.createdDate}
        error={touched.createdDate && errors.createdDate}
        value={values.createdDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <TextInput
        id="createdBy"
        label="Created By"
        helperText=""
        placeholder=""
        touched={touched.createdBy}
        error={touched.createdBy && errors.createdBy}
        value={values.createdBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <TextInput
        id="updatedDate"
        label="Updated Date"
        helperText=""
        placeholder=""
        touched={touched.updatedDate}
        error={touched.updatedDate && errors.updatedDate}
        value={values.updatedDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <TextInput
        id="updatedBy"
        label="Updated By"
        helperText=""
        placeholder=""
        touched={touched.updatedBy}
        error={touched.updatedBy && errors.updatedBy}
        value={values.updatedBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <Button type="submit" disabled={isSubmitting} variant="contained" color={'secondary'}>
        <SaveIcon />
        Save
      </Button>
    </div>
  </form>;
};

const formikEnhancer = withFormik({
  'validationSchema': Yup.object().shape({
    'name': Yup.string()
      .required('Name cannot be blank')
  }),
  'mapPropsToValues': (props) => props.initialFieldValues === null
    ? {
        'name': ''
      }
    : {
        'name': props.initialFieldValues.name,
        'createdDate': props.initialFieldValues.createdDate,
        'createdBy': props.initialFieldValues.userByCreatedById.email,
        'updatedDate': props.initialFieldValues.updatedDate,
        'updatedBy': props.initialFieldValues.userByUpdatedById.email
      },
  'handleSubmit': (payload, { props, setSubmitting }) => {
    setSubmitting(false);
    let variables;
    if (props.initialFieldValues === null) {
      variables = { 'indicatorGroup': payload };
    } else {
      variables = { 'indicatorGroupPatch': payload, 'id': props.initialFieldValues.id };
    }
    props.mutate({
      variables
    });
  },
  'displayName': 'IndicatorGroupForm'
});


const EnhancedIndicatorGroupForm = formikEnhancer(IndicatorGroupFormFields);

export default EnhancedIndicatorGroupForm;
