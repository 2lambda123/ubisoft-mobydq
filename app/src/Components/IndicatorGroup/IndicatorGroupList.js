import React from 'react';

import { connect } from 'react-redux';
import { setIndicatorGroupPage, setIndicatorGroupRowsPerPage, setIndicatorGroupRowTotal, setIndicatorGroupSortColumn } from './../../actions/indicatorGroupList';

import { Query } from 'react-apollo';
import IndicatorGroupRepository from './../../repository/IndicatorGroupRepository';
import { GraphQLError } from './../Error/GraphQLError';

import ListTable from '../ListTable/ListTable';
import LinkButton from './../../Components/FormInput/LinkButton';
import AddIcon from '@material-ui/icons/Add';

class IndicatorGroupList extends React.Component {
  render() {
    return (
      <Query
        query={IndicatorGroupRepository.getListPage()}
        variables={{ 'first': this.props.indicatorGroupRowsPerPage, 'offset': this.props.indicatorGroupPage * this.props.indicatorGroupRowsPerPage, 'orderBy': this.props.indicatorGroupSortColumn }}
        fetchPolicy={this.props.refetch ? 'cache-and-network' : 'cache-first'}
      >
        {({ loading, error, data }) => {
          if (loading) {
            return <p>Loading...</p>;
          }
          if (error) {
            return <GraphQLError error={error} />;
          }
          this.props.setIndicatorGroupRowTotal(data.allIndicatorGroups.totalCount);
          return (
            <div>
              <div style={{ 'float': 'left', 'marginLeft': '60px' }}>
                Indicator Groups
              </div>
              <div style={{ 'float': 'right' }}>
                <LinkButton label=<AddIcon /> type="create" color="secondary" variant="fab" to={'/indicator-group/new'}/>
              </div>
              <ListTable
                data={data.allIndicatorGroups.nodes}
                buttons={[
                  { 'function': 'execute', 'parameter': this._buildExecuteParam() },
                  { 'function': 'edit', 'parameter': '/indicator-group' },
                  { 'function': 'delete', 'parameter': this._buildDeleteParam() }
                ]}
                headerParams={this._buildHeaderParam()}
                footerParams={this._buildFooterParam()}
              />
            </div>
          );
        }}
      </Query>
    );
  }

  _buildExecuteParam() {
    return {
      'repository': IndicatorGroupRepository
    };
  }

  _buildDeleteParam() {
    return {
      'page': this.props.indicatorGroupPage,
      'rowTotal': this.props.indicatorGroupRowTotal,
      'rowsPerPage': this.props.indicatorGroupRowsPerPage,
      'setPage': this.props.setIndicatorGroupPage,
      'repository': IndicatorGroupRepository
    };
  }

  _buildFooterParam() {
    return {
      'page': this.props.indicatorGroupPage,
      'rowTotal': this.props.indicatorGroupRowTotal,
      'rowsPerPage': this.props.indicatorGroupRowsPerPage,
      'setPage': this.props.setIndicatorGroupPage,
      'setRowsPerPage': this.props.setIndicatorGroupRowsPerPage
    };
  }

  _buildHeaderParam() {
    return {
      'setSortField': this.props.setIndicatorGroupSortColumn,
      'sortField': this.props.indicatorGroupSortColumn
    };
  }
}

const mapStateToProps = (state) => ({
  'indicatorGroupPage': state.indicatorGroupPage,
  'indicatorGroupRowsPerPage': state.indicatorGroupRowsPerPage,
  'indicatorGroupRowTotal': state.indicatorGroupRowTotal,
  'indicatorGroupSortColumn': state.indicatorGroupSortColumn
});

const mapDispatchToProps = (dispatch) => ({
  'setIndicatorGroupPage': (page) => dispatch(setIndicatorGroupPage(page)),
  'setIndicatorGroupRowsPerPage': (rowsPerPage) => dispatch(setIndicatorGroupRowsPerPage(rowsPerPage)),
  'setIndicatorGroupRowTotal': (rowTotal) => dispatch(setIndicatorGroupRowTotal(rowTotal)),
  'setIndicatorGroupSortColumn': (sortColumn) => dispatch(setIndicatorGroupSortColumn(sortColumn))
});

export default connect(mapStateToProps, mapDispatchToProps)(IndicatorGroupList);
