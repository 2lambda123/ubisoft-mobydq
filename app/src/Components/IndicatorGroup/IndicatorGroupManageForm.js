import React from 'react';

export const IndicatorGroupManageForm = (props) => <div> Not yet implemented </div>;