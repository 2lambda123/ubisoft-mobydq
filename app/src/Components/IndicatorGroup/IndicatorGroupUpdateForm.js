import React from 'react';

import { Query } from 'react-apollo';
import IndicatorGroupRepository from '../../repository/IndicatorGroupRepository';
import GraphQLError from './../Error/GraphQLError';

import NotFoundComponent from '../Error/NotFoundComponent';

import { BaseForm } from '../Base/Form';
import EnhancedIndicatorGroupForm from './IndicatorGroupForm';

export const IndicatorGroupUpdateForm = ({ ...props }) => <Query query={IndicatorGroupRepository.display()} variables={{ 'id': props.match.params.id }}>
  {({ loading, error, data }) => {
    if (typeof IndicatorGroupRepository.display !== 'function') {
      throw new TypeError('Repository must implement update function.');
    }
    if (loading) {
      return <GraphQLError error={error}/>;
    }
    if (error) {
      return <p>Error :(</p>;
    }
    return (
      data.indicatorGroupById === null
        ? <NotFoundComponent/>
        : <BaseForm ComponentRepository={IndicatorGroupRepository} FormComponent={EnhancedIndicatorGroupForm}
          afterSaveRoute="/indicator-group/" title="Edit Indicator Group" initialFieldValues={data.indicatorGroupById} {...props}/>
    );
  }}
</Query>;
