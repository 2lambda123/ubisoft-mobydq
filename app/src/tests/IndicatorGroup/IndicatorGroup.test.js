import React from 'react';
import { shallowWrap } from './../../setupTests';

import IndicatorGroup from './../../Components/IndicatorGroup/IndicatorGroup';

describe('IndicatorGroup component unit test', () => {
  const match = {
    'url': '/indicator-group'
  };
  const wrapper = shallowWrap(<IndicatorGroup match={match}/>);
  it('matches snapshot', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('contains exactly 3 routes', () => {
    expect(wrapper.dive().find('Route')).toHaveLength(3);
  });
});
