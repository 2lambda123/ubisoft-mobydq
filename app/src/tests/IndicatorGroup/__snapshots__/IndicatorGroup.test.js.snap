// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorGroup component unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IndicatorGroup
    classes={
      Object {
        "appBar": "IndicatorGroup-appBar-4",
        "appBarShift": "IndicatorGroup-appBarShift-5",
        "appBarSpacer": "IndicatorGroup-appBarSpacer-11",
        "chartContainer": "IndicatorGroup-chartContainer-13",
        "content": "IndicatorGroup-content-12",
        "drawerPaper": "IndicatorGroup-drawerPaper-9",
        "drawerPaperClose": "IndicatorGroup-drawerPaperClose-10",
        "formContainer": "IndicatorGroup-formContainer-14",
        "menuButton": "IndicatorGroup-menuButton-6",
        "menuButtonHidden": "IndicatorGroup-menuButtonHidden-7",
        "root": "IndicatorGroup-root-1",
        "tableContainer": "IndicatorGroup-tableContainer-15",
        "title": "IndicatorGroup-title-8",
        "toolbar": "IndicatorGroup-toolbar-2",
        "toolbarIcon": "IndicatorGroup-toolbarIcon-3",
      }
    }
    match={
      Object {
        "url": "/indicator-group",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="IndicatorGroup-appBarSpacer-11"
        />,
        <WithStyles(Typography)
          className="IndicatorGroup-chartContainer-13"
          gutterBottom={true}
          variant="display1"
        >
          <Route
            component={[Function]}
            path="/indicator-group/new"
          />
          <Route
            component={[Function]}
            path="/indicator-group/edit/:id"
          />
          <Route
            exact={true}
            path="/indicator-group"
            render={[Function]}
          />
        </WithStyles(Typography)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "IndicatorGroup-appBarSpacer-11",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/indicator-group/new"
            />,
            <Route
              component={[Function]}
              path="/indicator-group/edit/:id"
            />,
            <Route
              exact={true}
              path="/indicator-group"
              render={[Function]}
            />,
          ],
          "className": "IndicatorGroup-chartContainer-13",
          "gutterBottom": true,
          "variant": "display1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/indicator-group/new",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/indicator-group/edit/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/indicator-group",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="IndicatorGroup-appBarSpacer-11"
          />,
          <WithStyles(Typography)
            className="IndicatorGroup-chartContainer-13"
            gutterBottom={true}
            variant="display1"
          >
            <Route
              component={[Function]}
              path="/indicator-group/new"
            />
            <Route
              component={[Function]}
              path="/indicator-group/edit/:id"
            />
            <Route
              exact={true}
              path="/indicator-group"
              render={[Function]}
            />
          </WithStyles(Typography)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "IndicatorGroup-appBarSpacer-11",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/indicator-group/new"
              />,
              <Route
                component={[Function]}
                path="/indicator-group/edit/:id"
              />,
              <Route
                exact={true}
                path="/indicator-group"
                render={[Function]}
              />,
            ],
            "className": "IndicatorGroup-chartContainer-13",
            "gutterBottom": true,
            "variant": "display1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/indicator-group/new",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/indicator-group/edit/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/indicator-group",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
}
`;
